ui/app/AppMain.qml

```diff
@@ -190,6 +190,11 @@ RowLayout {
         Layout.alignment: Qt.AlignLeft | Qt.AlignTop
         Layout.fillHeight: true
         currentIndex: tabBar.currentIndex
+        onCurrentIndexChanged: {
+            if (typeof this.children[currentIndex].onActivated === "function") {
+                this.children[currentIndex].onActivated()
+            }
+        }
```

Comment:
IMO this is a bit of a hack because there's no built-in "onActivate" hook in StackLayouts.. While this does work well, I have a feeling we're probably using a wrong Type here to show different views. Maybe `StackView` is what we actually want here?

Anyways, food for thought.

Reply:
Very fair point. I think you're right -- this should likely be a StackView. I'd be curious to know why StackLayout was chosen instead.

Reply:
The reason is we were just starting the project and didn't know anything lol